pkg/system/phase2_creating.go

```diff
@@ -92,8 +92,19 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 	if err := r.ReconcileObject(r.ServiceMgmt, r.SetDesiredServiceMgmt); err != nil {
 		return err
 	}
-	if err := r.ReconcileObject(r.ServiceDb, r.SetDesiredServiceDB); err != nil {
-		return err
+	if r.NooBaa.Spec.DBType == "postgres" {
+		if err := r.ReconcileObject(r.ServiceDbPg, r.SetDesiredServiceDB); err != nil {
```

Comment:
@dannyzaken not the same service. we need both the old one for mongo and new one for pg. so r.ServiceDbPg instead of r.ServiceDb. (if I understand your question right...)